synapse/util/caches/ttlcache.py

```diff
@@ -160,11 +160,11 @@ def expire(self) -> None:
 
 
 @attr.s(frozen=True, slots=True, auto_attribs=True)
-class _CacheEntry:  # Should be Generic[KT, VT]. See python-attrs/attrs#313
+class _CacheEntry(Generic[KT, VT]):
```

Comment:
Aside: It is very confusing that we have multiple classes with the same name.

Reply:
Yes, I try to avoid the caching machinery where I can!